bestie/server/service.go

```diff
@@ -97,3 +139,38 @@ func (s *serviceStats) incrementEventsTotal(bevid interface{}) {
 	}
 	s.eventsTotal.WithLabelValues(label).Inc()
 }
+
+// BigQuery operational exceptions.
+func (s *serviceStats) incrementBigQueryExcessDelay() {
+	s.bigQueryExceptionTotal.WithLabelValues("excess_delay").Inc()
+}
+func (s *serviceStats) incrementBigQueryProtobufError() {
+	s.bigQueryExceptionTotal.WithLabelValues("protobuf_error").Inc()
+}
+func (s *serviceStats) incrementBigQueryTableNotFound() {
+	s.bigQueryExceptionTotal.WithLabelValues("table_not_found").Inc()
+}
+
+// BigQuery row insertion delay (histogram).
+func (s *serviceStats) incrementBigQueryInsertDelay(seconds int) {
+	s.bigQueryInsertDelay.Observe(float64(seconds))
+}
+
+// BigQuery row insertion outcome.
+func (s *serviceStats) incrementBigQueryInsertError() {
+	s.bigQueryInsertTotal.WithLabelValues("error").Inc()
+}
+func (s *serviceStats) incrementBigQueryInsertOK() {
+	s.bigQueryInsertTotal.WithLabelValues("ok").Inc()
+}
+func (s *serviceStats) incrementBigQueryInsertTimeout() {
+	s.bigQueryInsertTotal.WithLabelValues("timeout").Inc()
+}
+
+// BigQuery total metrics processed.
+func (s *serviceStats) incrementBigQueryMetricDiscard(count int) {
+	s.bigQueryMetricTotal.WithLabelValues("discard").Add(float64(count))
+}
+func (s *serviceStats) incrementBigQueryMetricUpload(count int) {
+	s.bigQueryMetricTotal.WithLabelValues("upload").Add(float64(count))
+}
```

Comment:
I wanted to abstract away the specific details of the Prometheus metric API from the rest of the code. I've now changed this completely and instead of having individual accessor methods, I defined a couple of general functions that are called throughout the code using a counter ID. These in turn invoke a common function as a goroutine (per Adam's comments above) to update the actual Prometheus counter.